src/hotspot/share/classfile/placeholders.hpp

```diff
@@ -142,8 +142,12 @@ class PlaceholderEntry : public HashtableEntry<Symbol*, mtClass> {
 
   Symbol*            supername()           const { return _supername; }
   void               set_supername(Symbol* supername) {
-    _supername = supername;
-    if (_supername != NULL) _supername->increment_refcount();
+    if (supername != _supername) {
```

Comment:
Why isn't there an if (_supername == NULL) _supername->decrement_refcount(); before assigning the new supername?

Reply:
Also it turn out that checking for NULL may be not enough. PlaceholderEntry has no constructor, so members can contain garbage.

Reply:
I see.  This is the old hashtable implementation.

Reply:
Coleen, thanks for the review.